File: web/src/api/mine/mine_books.js
import service from '@/utils/request'

export const createMineBooks = (data) => {
  return service({
    url: '/mineBooks/createMineBooks',
    method: 'post',
    data
  })
}

export const deleteMineBooks = (params) => {
  return service({
    url: '/mineBooks/deleteMineBooks',
    method: 'delete',
    params
  })
}

export const deleteMineBooksByIds = (params) => {
  return service({
    url: '/mineBooks/deleteMineBooksByIds',
    method: 'delete',
    params
  })
}

export const updateMineBooks = (data) => {
  return service({
    url: '/mineBooks/updateMineBooks',
    method: 'put',
    data
  })
}

export const findMineBooks = (params) => {
  return service({
    url: '/mineBooks/findMineBooks',
    method: 'get',
    params
  })
}

export const getMineBooksList = (params) => {
  return service({
    url: '/mineBooks/getMineBooksList',
    method: 'get',
    params
  })
}

File: server/router/mine/mine_books.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineBooksRouter struct{}

func (s *MineBooksRouter) InitMineBooksRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineBooksRouter := Router.Group("mineBooks").Use(middleware.OperationRecord())
	mineBooksRouterWithoutRecord := Router.Group("mineBooks")
	mineBooksRouterWithoutAuth := PublicRouter.Group("mineBooks")
	{
		mineBooksRouter.POST("createMineBooks", mineBooksApi.CreateMineBooks)
		mineBooksRouter.DELETE("deleteMineBooks", mineBooksApi.DeleteMineBooks)
		mineBooksRouter.DELETE("deleteMineBooksByIds", mineBooksApi.DeleteMineBooksByIds)
		mineBooksRouter.PUT("updateMineBooks", mineBooksApi.UpdateMineBooks)
	}
	{
		mineBooksRouterWithoutRecord.GET("findMineBooks", mineBooksApi.FindMineBooks)
		mineBooksRouterWithoutRecord.GET("getMineBooksList", mineBooksApi.GetMineBooksList)
	}
	{
		_ = mineBooksRouterWithoutAuth
	}
}
